subiquity/server/controllers/cmdlist.py

```diff
@@ -78,6 +79,16 @@ async def run(self, context):
         env = self.env()
         for i, cmd in enumerate(tuple(self.builtin_cmds) + tuple(self.cmds)):
             desc = cmd.desc()
+
+            # If the path to the command isn't found on the snap we should
+            # drop the snap specific environment variables. If the command
+            # is `env`, we should let the user manage the environment
+            command = shlex.split(desc)[0]
+            path = shutil.which(command)
+            if path is not None:
+                if not path.startswith("/snap") and command != "env":
```

Comment:
If we're special casing `env`, someone might have specified the full path, which means all of `("env", "/bin/env", "/usr/bin/env")` are possible.

Reply:
Hmmm, that's a good point. My thought was if someone passed their command via `env` they probably don't want us stripping everything for them (unless of course they're passing `env -i`). Should we special case on all three of these options, or just not special case at all?

Reply:
/usr/bin/env does link to libc6, so the better answer is orig_env, I think.
for `env -i`  I think it doesn't matter what is done here.

> if someone passed their command via env they probably don't want us stripping everything for them

I would also prefer to not break someone's existing autoinstall, but if they are setting environment variables today I think the common scenarios are:
* they have a special case that is not covered by the above
* they already ran into this problem and are working around it, and giving a clean environment should still be ok

So I think the answer is to not special case env.  If you can think of a corner case not covered in this plan please point it out.

Reply:
I can't think of any other cases. That covers my concerns, so I'll remove the special casing. Thanks!